runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -7450,29 +7506,32 @@ void TR_EscapeAnalysis::heapifyBeforeColdBlocks(Candidate *candidate)
            initStoreTree->join(nextToStart);
            }
```

Comment:
In an off-line conversation, Andrew @andrewcraik recommended that I try to pull the code that appears between the two blocks of code guarded by tests of `useTernaryOp` into a separate method, which would then be called in both cases. That would allow the two blocks of code for the `!useTernaryOp` case to be merged, improving readability.

Andrew, I've made that change in commit https://github.com/eclipse/openj9/pull/8364/commits/f66b989d4f9d4b3335a84fb5723074da03ab862f.  If you are OK with that change, I will merge the two commits.

Reply:
@hzongaro thank you for having a look at that - I think the code is much cleaner now - no locals carried across ifs and the isolated logic actually has an nice understandable name and makes a lot of sense. Overall a huge improvement - thank you for doing the work to add another layer of polish to this change. Please squash and I will do a final read and start sanity.